planner/core/stats.go

```diff
@@ -549,6 +541,12 @@ func (ds *DataSource) buildIndexMergeOrPath(partialPaths []*util.AccessPath, cur
 	indexMergePath := &util.AccessPath{PartialIndexPaths: partialPaths}
 	indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[:current]...)
 	indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[current+1:]...)
+	for _, path := range partialPaths {
+		if len(path.TableFilters) > 0 {
+			indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[current])
+			break
+		}
+	}
```

Comment:
Theoretically, we should update the estimated row count of `TableRowIDScan` and `Selection` for this index merge plan as well, though it is pretty hard to figure out the row count. See also function `buildIndexMergeTableScan`. @time-and-fate any thoughts?

Reply:
I think we can use all `partialPath.AccessConds` instead of currently the entire DNF condition to calculate selectivity in `generateIndexMergeOrPaths()`. 
As for the `Selection` above the `TableRowIDScan`, I think we can choose not to modify it, but I wonder if using `ds.stats` directly instead of the current `selectivity(tableFilters) * totalRowCount`  could be better in `buildIndexMergeTableScan()`.

Reply:
Updated, PTAL. @time-and-fate @eurekaka